lib/thesis/api_controller.ex

```diff
@@ -18,6 +18,30 @@ defmodule Thesis.ApiController do
     json conn, %{}
   end
 
+  def upload_file(conn, %{"file" => ""}), do: json conn, %{path: ""}
+  def upload_file(conn, %{"file" => file}) do
+    path = uploader.upload(file)
+    json conn, %{path: path}
+  end
+  def upload_file(conn, _), do: json conn, %{path: ""}
```

Comment:
This is simple, but we should really pass back errors if they occur. The `uploader.upload/1` function should return a tuple with `{:ok, path}` or `{:error, reason}`.